outbox.js

```diff
@@ -1,310 +1,242 @@
-    this.page = parseInt(this.getAttribute("page"), 10) || this.page;
-    this.loadOutbox(this.getAttribute("url"));
+      parseInt(this.getAttribute('num-posts'), 10) || this.numPosts
+    this.page = parseInt(this.getAttribute('page'), 10) || this.page
+    this.loadOutbox(this.getAttribute('url'))
   }
 
-  async loadOutbox(outboxUrl) {
-    this.clearContent();
+  async loadOutbox (outboxUrl) {
+    this.clearContent()
     for await (const item of this.fetchOutboxItems(outboxUrl)) {
-      this.processItem(item);
+      this.processItem(item)
     }
   }
 
-  processItem(item) {
-    const itemKey = item.id || item.object;
+  processItem (item) {
+    const itemKey = item.id || item.object
     if (!itemKey) {
-      console.error("Item key is undefined, item:", item);
-      return;
+      console.error('Item key is undefined, item:', item)
+      return
     }
     if (!this.renderedItems.has(itemKey)) {
-      this.renderItem(item);
-      this.renderedItems.set(itemKey, true);
+      this.renderItem(item)
+      this.renderedItems.set(itemKey, true)
     }
   }
 
-  async *fetchOutboxItems(outboxUrl) {
+  async * fetchOutboxItems (outboxUrl) {
     if (!outboxUrl) {
-      console.error("No outbox URL provided");
-      return;
+      console.error('No outbox URL provided')
+      return
     }
 
-    try {
-      let response;
-      // Check the scheme and adjust the URL for unsupported schemes before fetching
-      if (outboxUrl.startsWith("hyper://")) {
-        const gatewayUrl = outboxUrl.replace(
-          "hyper://",
-          "https://hyper.hypha.coop/hyper/"
-        );
-        response = await fetch(gatewayUrl);
-      } else if (outboxUrl.startsWith("ipns://")) {
-        const gatewayUrl = outboxUrl.replace(
-          "ipns://",
-          "https://ipfs.hypha.coop/ipns/"
-        );
-        response = await fetch(gatewayUrl);
-      } else {
-        response = await fetch(outboxUrl);
-      }
-
-      if (!response.ok) {
-        throw new Error(`HTTP error! Status: ${response.status}`);
-      }
-      const outbox = await response.json();
-
-      // Adjust for both direct items and items loaded via URLs
-      const items = [];
-      for (const itemOrUrl of outbox.orderedItems) {
-        if (typeof itemOrUrl === "string") {
-          // URL to an activity
-          const itemResponse = await fetch(itemOrUrl);
-          if (itemResponse.ok) {
-            const item = await itemResponse.json();
-            items.push(item);
-          }
-        } else {
-          items.push(itemOrUrl); // Directly included activity
-        }
-      }
-
-      this.totalPages = Math.ceil(items.length / this.numPosts);
-      this.page = Math.min(this.page, this.totalPages);
+    /*
+    this.totalPages = Math.ceil(items.length / this.numPosts);
+    this.page = Math.min(this.page, this.totalPages);
 
       // Calculate the range of items to be loaded based on the current page and numPosts
```

Comment:
Mind fixing the pagination by having a `loadMore` button which will use paging to load more items when possible?